### PR TITLE
Add performance scripts for the GDPR, PCI-DSS & NIST modules

### PR DESCRIPTION
**Pull request by mauceballos**
add test cases for GDPR && PCI-DSS Modules and your tabs
* Dashboard
* Control
* Events

### CODE REVIEW

File: test/performance/modules/GDPR-module-dashboard.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  try {
+    // Click on Kibana menu to access Wazuh App link
+    await commands.wait.bySelector('button[data-test-subj="toggleNavButton"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('button[data-test-subj="toggleNavButton"]')
+    await commands.wait.bySelector('a[href$="/app/wazuh"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('a[href$="/app/wazuh"]')
+    
+    //Wait for an Wazuh home page component to be loaded
+    await commands.wait.byXpath('//*[contains(@class,"euiTitle euiTitle--small euiCard__title")]//*[contains(text(),"GDPR")]', WAIT_TIMEOUT)
+    await commands.wait.byCondition("!isNaN(parseInt(document.querySelector('.statWithLink').innerHTML))", WAIT_TIMEOUT)
```
Comment (Mayons95) [inline test/performance/modules/GDPR-module-dashboard.js:17]: This should be in the same line

File: test/performance/modules/GDPR-module-events.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  try {
+    // Click on Kibana menu to access Wazuh App link
+    await commands.wait.bySelector('button[data-test-subj="toggleNavButton"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('button[data-test-subj="toggleNavButton"]')
+    await commands.wait.bySelector('a[href$="/app/wazuh"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('a[href$="/app/wazuh"]')
+    
+    //Wait for an Wazuh home page component to be loaded
+    await commands.wait.byXpath('//*[contains(@class,"euiTitle euiTitle--small euiCard__title")]//*[contains(text(),"GDPR")]', WAIT_TIMEOUT)
+    await commands.wait.byCondition("!isNaN(parseInt(document.querySelector('.statWithLink').innerHTML))", WAIT_TIMEOUT)
```
Comment (Mayons95) [inline test/performance/modules/GDPR-module-events.js:17]: This should be in the same line

File: test/performance/modules/NIST-module-controls.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  try {
+    // Click on Kibana menu to access Wazuh App link
+    await commands.wait.bySelector('button[data-test-subj="toggleNavButton"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('button[data-test-subj="toggleNavButton"]')
+    await commands.wait.bySelector('a[href$="/app/wazuh"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('a[href$="/app/wazuh"]')
+    
+    //Wait for an Wazuh home page component to be loaded
+    await commands.wait.byXpath('//*[contains(@class,"euiTitle euiTitle--small euiCard__title")]//*[contains(text(),"NIST 800-53")]', WAIT_TIMEOUT)
+    await commands.wait.byCondition("!isNaN(parseInt(document.querySelector('.statWithLink').innerHTML))", WAIT_TIMEOUT)
```
Comment (Mayons95) [inline test/performance/modules/NIST-module-controls.js:17]: This should be in the same line

File: test/performance/modules/NIST-module-dashboard.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  try {
+    // Click on Kibana menu to access Wazuh App link
+    await commands.wait.bySelector('button[data-test-subj="toggleNavButton"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('button[data-test-subj="toggleNavButton"]')
+    await commands.wait.bySelector('a[href$="/app/wazuh"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('a[href$="/app/wazuh"]')
+    
+    //Wait for an Wazuh home page component to be loaded
+    await commands.wait.byXpath('//*[contains(@class,"euiTitle euiTitle--small euiCard__title")]//*[contains(text(),"NIST 800-53")]', WAIT_TIMEOUT)
+    await commands.wait.byCondition("!isNaN(parseInt(document.querySelector('.statWithLink').innerHTML))", WAIT_TIMEOUT)
```
Comment (Mayons95) [inline test/performance/modules/NIST-module-dashboard.js:17]: This should be in the same line

File: test/performance/modules/NIST-module-events.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  try {
+    // Click on Kibana menu to access Wazuh App link
+    await commands.wait.bySelector('button[data-test-subj="toggleNavButton"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('button[data-test-subj="toggleNavButton"]')
+    await commands.wait.bySelector('a[href$="/app/wazuh"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('a[href$="/app/wazuh"]')
+    
+    //Wait for an Wazuh home page component to be loaded
+    await commands.wait.byXpath('//*[contains(@class,"euiTitle euiTitle--small euiCard__title")]//*[contains(text(),"NIST 800-53")]', WAIT_TIMEOUT)
+    await commands.wait.byCondition("!isNaN(parseInt(document.querySelector('.statWithLink').innerHTML))", WAIT_TIMEOUT)
```
Comment (Mayons95) [inline test/performance/modules/NIST-module-events.js:17]: This should be in the same line

File: test/performance/modules/PCI-DSS-module-controls.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  try {
+    // Click on Kibana menu to access Wazuh App link
+    await commands.wait.bySelector('button[data-test-subj="toggleNavButton"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('button[data-test-subj="toggleNavButton"]')
+    await commands.wait.bySelector('a[href$="/app/wazuh"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('a[href$="/app/wazuh"]')
+    
+    //Wait for an Wazuh home page component to be loaded
+    await commands.wait.byXpath('//*[contains(@class,"euiTitle euiTitle--small euiCard__title")]//*[contains(text(),"PCI DSS")]', WAIT_TIMEOUT)
+    await commands.wait.byCondition("!isNaN(parseInt(document.querySelector('.statWithLink').innerHTML))", WAIT_TIMEOUT)
```
Comment (Mayons95) [inline test/performance/modules/PCI-DSS-module-controls.js:17]: This should be in the same line

File: test/performance/modules/PCI-DSS-module-dashboard.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  try {
+    // Click on Kibana menu to access Wazuh App link
+    await commands.wait.bySelector('button[data-test-subj="toggleNavButton"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('button[data-test-subj="toggleNavButton"]')
+    await commands.wait.bySelector('a[href$="/app/wazuh"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('a[href$="/app/wazuh"]')
+    
+    //Wait for an Wazuh home page component to be loaded
+    await commands.wait.byXpath('//*[contains(@class,"euiTitle euiTitle--small euiCard__title")]//*[contains(text(),"PCI DSS")]', WAIT_TIMEOUT)
+    await commands.wait.byCondition("!isNaN(parseInt(document.querySelector('.statWithLink').innerHTML))", WAIT_TIMEOUT)
```
Comment (Mayons95) [inline test/performance/modules/PCI-DSS-module-dashboard.js:17]: This should be in the same line

File: test/performance/modules/PCI-DSS-module-events.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  try {
+    // Click on Kibana menu to access Wazuh App link
+    await commands.wait.bySelector('button[data-test-subj="toggleNavButton"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('button[data-test-subj="toggleNavButton"]')
+    await commands.wait.bySelector('a[href$="/app/wazuh"]', WAIT_TIMEOUT)
+    await commands.click.bySelector('a[href$="/app/wazuh"]')
+    
+    //Wait for an Wazuh home page component to be loaded
+    await commands.wait.byXpath('//*[contains(@class,"euiTitle euiTitle--small euiCard__title")]//*[contains(text(),"PCI DSS")]', WAIT_TIMEOUT)
+    await commands.wait.byCondition("!isNaN(parseInt(document.querySelector('.statWithLink').innerHTML))", WAIT_TIMEOUT)
```
Comment (Mayons95) [inline test/performance/modules/PCI-DSS-module-events.js:17]: This should be in the same line

**Review by Mayons95 — CHANGES_REQUESTED** [review]
See the comments on the files

**Review by Mayons95 — APPROVED** [review]

File: test/performance/modules/GDPR-module-dashboard.js (hunk 1 above)
Comment (mauceballos) [reply test/performance/modules/GDPR-module-dashboard.js:17]: mmm, nop, shouldn't.

**Comment by mauceballos** [comment]
**GDPR**  Module

>       - Control
>       - Dashboard
>       - Events
>       
> <details>
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-GDPR-module-controls-js-2022-08-08-10-25-49-pages-html-2022-08-08-10_29_07
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-GDPR-module-controls-js-2022-08-08-10-25-49-pages-localhost-GDPR-module-Controls-index-html-2022-08-08-10_29_22
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-GDPR-module-controls-js-2022-08-08-10-25-49-pages-localhost-GDPR-module-Dashboard-index-html-2022-08-08-10_29_38
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-GDPR-module-controls-js-2022-08-08-10-25-49-pages-localhost-GDPR-module-Events-index-html-2022-08-08-10_29_52
> 
>  </details>



**PCI-DSS**  Module

>       - Control
>       - Dashboard
>       - Events
> 
> <details>
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-PCI-DSS-module-controls-js-2022-08-08-10-47-01-pages-html-2022-08-08-10_50_47
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-PCI-DSS-module-controls-js-2022-08-08-10-47-01-pages-localhost-PCI-DSS-module-Controls-index-html-2022-08-08-10_50_38
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-PCI-DSS-module-controls-js-2022-08-08-10-47-01-pages-localhost-PCI-DSS-module-Dashboard-index-html-2022-08-08-10_50_28
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-PCI-DSS-module-controls-js-2022-08-08-10-47-01-pages-localhost-PCI-DSS-module-Events-index-html-2022-08-08-10_50_15
> 
>  </details>

 

**NIST** Module

>       - Control
>       - Dashboard
>       - Events
> 
> <details>
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-NIST-module-controls-js-2022-08-08-10-36-39-pages-html-2022-08-08-10_40_14
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-NIST-module-controls-js-2022-08-08-10-36-39-pages-localhost-NIST-800-53-module-Controls-index-html-2022-08-08-10_40_37
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-NIST-module-controls-js-2022-08-08-10-36-39-pages-localhost-NIST-800-53-module-Dashboard-index-html-2022-08-08-10_40_51
> 
> screencapture-file-home-mauri-repoPerformance-wazuh-kibana-app-test-performance-modules-sitespeed-result-NIST-module-controls-js-2022-08-08-10-36-39-pages-localhost-NIST-800-53-module-Events-index-html-2022-08-08-10_41_03
> 
>  </details>

**Pull request closed without merging**